controllers/activemqartemis_controller2_test.go

```diff
@@ -0,0 +1,329 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+/*
+As usual, we start with the necessary imports. We also define some utility variables.
+*/
+package controllers
+
+import (
+	"container/list"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/artemiscloud/activemq-artemis-operator/pkg/resources/volumes"
+	"github.com/artemiscloud/activemq-artemis-operator/pkg/utils/common"
+	"github.com/artemiscloud/activemq-artemis-operator/pkg/utils/namer"
+
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/watch"
+
+	brokerv1beta1 "github.com/artemiscloud/activemq-artemis-operator/api/v1beta1"
+)
+
+var _ = Describe("artemis controller 2", func() {
+
+	// see what has changed from the controllers perspective, what we watch
+	toWatch := []client.ObjectList{&brokerv1beta1.ActiveMQArtemisList{}, &appsv1.StatefulSetList{}, &corev1.PodList{}}
+	wis := list.New()
+	BeforeEach(func() {
+
+		BeforeEachSpec()
+
+		if verbose {
+			fmt.Println("Time with MicroSeconds: ", time.Now().Format("2006-01-02 15:04:05.000000"), " test:", CurrentSpecReport())
+		}
+
+		for _, li := range toWatch {
+
+			wc, err := client.NewWithWatch(testEnv.Config, client.Options{})
+			if err != nil {
+				fmt.Printf("Err on watch client:  %v\n", err)
+				return
+			}
+
+			// see what changed
+			wi, err := wc.Watch(ctx, li, &client.ListOptions{})
+			if err != nil {
+				fmt.Printf("Err on watch:  %v\n", err)
+			}
+			wis.PushBack(wi)
+
+			go func() {
+				for event := range wi.ResultChan() {
+					if verbose {
+						fmt.Printf("%v : Object: %v\n", event.Type, event.Object)
+					}
+				}
+			}()
+		}
+	})
+
+	AfterEach(func() {
+		for e := wis.Front(); e != nil; e = e.Next() {
+			e.Value.(watch.Interface).Stop()
+		}
+
+		AfterEachSpec()
+	})
+
+	Context("persistent volumes tests", Label("controller-2-test"), func() {
+		It("external volumes attach", func() {
+			if os.Getenv("USE_EXISTING_CLUSTER") == "true" {
+
+				By("create extra volume")
+				pvc, createdPvc := DeployCustomPVC("my-pvc", defaultNamespace, func(candidate *corev1.PersistentVolumeClaim) {
+
+					candidate.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteMany,
```

Comment:
Running the full test-suite on an AWS OpenShift cluster causes the failure of the test `external volumes attach` because the default storage class doesn't support the access mode `ReadWriteMany`. That test failure could be easily fixed using the access mode `ReadWriteOnce`, i.e.
```suggestion
						corev1.ReadWriteOnce,
```

Reply:
ok will update. Thanks!